CIP-0155/README.md

```diff
@@ -0,0 +1,136 @@
+
+Decentralised protocol developers SHOULD submit proposals to the SRV Prefix Registry, so SPOs, who deploy them, can have an authoritative information how to do it.
+
+## Specification
+
+### SRV Prefix Registry
+
+The registry is available in `JSON` format [registry.json].
+
+### SRV Prefix Semantics
+
+When a **cardano** node implementation reads an SRV record from a ledger, it must add the _cardano_ prefix from the table above before making a DNS lookup, e.g. it should do a DNS query for `_cardano._tcp.<srv-record-from-ledger>`.
+
+This design allows decentralised protocols to use SRV records registered in the ledger for different purposes, e.g. a **mithril** node can use them to learn about end-points of its network.
+
+Each prefix SHOULD start with `_cardano._tcp` or `_cardano._udp`, to avoid clashes with services not related to `Cardano`.
+
+#### SRV Registry Rules
+
+* Each decentralised protocol can have at most one entry in the registry.
+* A CIP process assigns new entries to [registry.json], after a careful consideration and consultation with all the involved parties (see #acceptance-criteria below).
+* Entries cannot be removed, but can be revoked by assigning a `Revoked` status.
+  This can only happen if a decentralised protocol is no longer supported.
+
+### Example
+
+When registering a cardano pool on `example.com` domain using an `SRV` record, one should use:
+```shell
+cardano-cli latest stake-pool ... --multi-host-pool-relay example.com
+```
+(see [register-stake-pool]); and configure SRV record at `_cardano._tcp.example.com` to point to **Cardano** relays, `_mithril._tcp.example.com` to point to **Mithril** relays (see [srv], currently under development).
+
+A **Cardano** node implementation, when retrieving information from a registration certificate from the ledger, will receive `example.com`, and it must prepend the `_cardano._tcp` prefix to make an SRV query.  Such a query might return:
+
+```
+_cardano._tcp.example.com 86400 IN SRV 10 5 3001 cardano.example.com
+```
+From this, we learn that a Cardano node is available on port `3001` on IPs resolved by a regular DNS query to `cardano.example.com`.
+Refer to the [Cloudflare documentation][srv] for a deeper understanding of other fields.
+
+
+## Rationale: how does this CIP achieve its goals?
+
+This CIP constructs a process to maintain SRV registry, and thus provides authritative information for SPOs and decentralised protocol developers.
+
+
+## Path to Active
+
+### Acceptance Criteria
+
+The CIP should be accepted by:
+
+* [ ] [**Technical Steering Committee**][tsc]
+
+And when there's no major objection from one of the currently involved parties:
+
+* [ ] [**Amaru Team**][amaru] accepts the proposal
+* [ ] [**Cardano-Node Team**][cardano-node] accepts the proposal
+* [ ] [**Gouroboros Team**][gouroboros] accepts the proposal
+* [ ] [**Hydra Team**][hydra] accepts the proposal
+* [ ] [**Mithril Team**][mithril] accepts the proposal
```

Comment:
```suggestion
* [x] [**Cardano-Node Team**][cardano-node] accepts the proposal
* [ ] [**Amaru Team**][amaru] accepts the proposal
* [ ] [**Gouroboros Team**][gouroboros] accepts the proposal
* [ ] [**Hydra Team**][hydra] accepts the proposal
* [ ] [**Mithril Team**][mithril] accepts the proposal
```
I assume the proposal must at least have been "accepted" if already [released in `cardano-node-10.6`](https://github.com/cardano-foundation/CIPs/pull/1033#discussion_r2242501265).